The job **`x86_64-gnu-llvm-17`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9037108394/job/24835442068) [(plain)](https://github.com/rust-lang/rust/commit/06beef4c6a3ac0dfcc4139f966e0488018fb515c/checks/24835442068/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 29.0s done
#16 DONE 34.4s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-17]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-17', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-17/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
   Compiling unwind v0.0.0 (/checkout/library/unwind)
error[E0599]: no method named `clone` found for type parameter `T` in the current scope
    --> library/alloc/src/rc.rs:1846:56
     |
1814 | impl<T, A: Allocator> Rc<T, A> {
     |      - method `clone` not found for this type parameter
...
1846 |         Rc::try_unwrap(this).unwrap_or_else(|rc| (*rc).clone())
     |                                                        ^^^^^ method not found in `T`
     = help: items from traits can only be used if the type parameter is bounded by the trait
     = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `clone`, perhaps you need to restrict type parameter `T` with it:
     |
1814 | impl<T: Clone, A: Allocator> Rc<T, A> {

error[E0599]: no method named `clone` found for type parameter `T` in the current scope
    --> library/alloc/src/sync.rs:2264:59
     |
     |
2232 | impl<T, A: Allocator> Arc<T, A> {
     |      - method `clone` not found for this type parameter
...
2264 |         Arc::try_unwrap(this).unwrap_or_else(|arc| (*arc).clone())
     |                                                           ^^^^^ method not found in `T`
     = help: items from traits can only be used if the type parameter is bounded by the trait
     = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `clone`, perhaps you need to restrict type parameter `T` with it:
     |
2232 | impl<T: Clone, A: Allocator> Arc<T, A> {

For more information about this error, try `rustc --explain E0599`.
error: could not compile `alloc` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

--- a/library/alloc/src/rc.rs
+++ b/library/alloc/src/rc.rs
@@ -1809,7 +1809,9 @@ impl<T: Clone, A: Allocator + Clone> Rc<T, A> {
         // reference to the allocation.
         unsafe { &mut this.ptr.as_mut().value }
     }
+}
 
+impl<T: Clone, A: Allocator> Rc<T, A> {
     /// If we have the only reference to `T` then unwrap it. Otherwise, clone `T` and return the
     /// clone.
     ///

--- a/library/alloc/src/sync.rs
+++ b/library/alloc/src/sync.rs
@@ -2227,7 +2227,9 @@ impl<T: Clone, A: Allocator + Clone> Arc<T, A> {
         // either unique to begin with, or became one upon cloning the contents.
         unsafe { Self::get_mut_unchecked(this) }
     }
+}
 
+impl<T: Clone, A: Allocator> Arc<T, A> {
     /// If we have the only reference to `T` then unwrap it. Otherwise, clone `T` and return the
     /// clone.
     ///

